map.js

```diff
@@ -8,6 +8,63 @@ const countries = mapjs.getAttribute('data-countries') || 'us,ca';
 
 mapboxgl.accessToken = mapjs.getAttribute('data-token');
 
+// for the layer filter, we have a vue component, which is managing
+// a mapbox control which we define (https://www.mapbox.com/mapbox-gl-js/api/#icontrol)
+
+// this is the vue component - it is passed the layers, and it creates
+// checkbox for each layer. We model the checks with an array of
+// strings, and when a box changes, we emit 'layer-change', which is
+// caught by the parent vue app, and processes the hiding and showing
+Vue.component('ragtag-layerfilter', {
+  props: ['layers'],
+  template: 
+'<div class="map-widget filter-widget">' +
+'  <template v-for="layer in layers">' +
+'    <div>' +
+'      <input' +
+'        type="checkbox"' +
+'        :value="layer.layerId"' +
+'        :id="layer.layerId"' +
+'        v-model="checkedLayers"' +
+'        @change="showHideLayers()"' +
+'      >' +
+'      <label for="layer.layerId"><img :src="\'assets/\'+layer.icon">{{layer.label}}</label>' +
+'    </div>' +
+'  </template>' +
+'</div>'
+,
```

Comment:
I am not a big fan of string-concatenated html in JavaScript; what about putting the html in events.html as a template:

```
<script type="text/x-template" id="layerfilter-template">
  <div class="map-widget filter-widget">
...
  </div>
</script>
```

and then referencing from the component
```
template: '#layerfilter-template',
```